core/server/master/src/main/java/alluxio/master/file/LostFileDetector.java

```diff
@@ -42,6 +44,8 @@
   public LostFileDetector(FileSystemMaster fileSystemMaster, InodeTree inodeTree) {
     mFileSystemMaster = fileSystemMaster;
     mInodeTree = inodeTree;
+    MetricsSystem.registerGaugeIfAbsent(MetricKey.MASTER_LOST_FILE_COUNT.getName(),
+        mFileSystemMaster::getLostFiles);
```

Comment:
getLostFiles will iterate all the lost blocks and get a list of lost file ids, which if register as gauge is very inefficient.
besides, are you trying to use getLostFiles().size() here?
i will prefer to use cached guaged with a relatively long cached period

Reply:
+1